wallet/src/main/scala/org/bitcoins/wallet/Wallet.scala

```diff
@@ -3,255 +3,41 @@ package org.bitcoins.wallet
-    utxoDAO.findAllUTXOs()
-
-  /**
-    * @param account Account to generate address from
-    * @param chainType What chain do we generate from? Internal change vs. external
-    */
-  private def getNewAddressHelper(
-      account: AccountDb,
-      chainType: HDChainType
-  ): Future[BitcoinAddress] = {
-
-    val accountIndex = account.hdAccount.index
-
-    val lastAddrOptF = chainType match {
-      case HDChainType.External =>
-        addressDAO.findMostRecentExternal(accountIndex)
-      case HDChainType.Change =>
-        addressDAO.findMostRecentChange(accountIndex)
-    }
-
-    lastAddrOptF.flatMap { lastAddrOpt =>
-      val addrPath: HDPath = lastAddrOpt match {
-        case Some(addr) =>
-          addr.path.next
-        case None =>
-          val account = HDAccount(DEFAULT_HD_COIN, accountIndex)
-          val chain = account.toChain(chainType)
-          val address = HDAddress(chain, 0)
-          address.toPath
-      }
-
-      val addressDb =
-        addrPath match {
-          case segwitPath: SegWitHDPath =>
-            val pathDiff = account.hdAccount.diff(segwitPath) match {
-              case Some(value) => value
-              case None =>
-                throw new RuntimeException(
-                  s"Could not diff ${account.hdAccount} and $segwitPath")
-            }
-
-            val pubkey = account.xpub.deriveChildPubKey(pathDiff) match {
-              case Failure(exception) => throw exception
-              case Success(value)     => value.key
-            }
-
-            AddressDbHelper
-              .getP2WPKHAddress(pubkey, segwitPath, networkParameters)
-          case _: HDPath =>
-            throw new IllegalArgumentException(
-              "P2PKH and nested segwit P2PKH not yet implemented")
-        }
-      val writeF = addressDAO.create(addressDb)
-      writeF.foreach { written =>
-        logger.info(
-          s"Got ${chainType} address ${written.address} at key path ${written.path} with pubkey ${written.ecPublicKey}")
-      }
-
-      writeF.map(_.address)
-    }
-  }
-
-  /**
-    * right now only generates P2WPKH addresses
-    *
-    * @inheritdoc
-    */
-  override def getNewAddress(account: AccountDb): Future[BitcoinAddress] = {
-    getNewAddressHelper(account, HDChainType.External)
-  }
-
-  /** Generates a new change address */
-  private def getNewChangeAddress(
-      account: AccountDb): Future[BitcoinAddress] = {
-    getNewAddressHelper(account, HDChainType.Change)
-  }
-
-  /** @inheritdoc */
-  override protected[wallet] def getDefaultAccount(): Future[AccountDb] = {
-    for {
-      account <- accountDAO.read((DEFAULT_HD_COIN, 0))
-    } yield
-      account.getOrElse(
-        throw new RuntimeException(
-          s"Could not find account with ${DEFAULT_HD_COIN.purpose.constant} " +
-            s"purpose field and ${DEFAULT_HD_COIN.coinType.toInt} coin field"))
+    WalletStorage.writeMnemonicToDisk(encrypted)
```

Comment:
can this fail? If it can, it doesn't look like it is represented in the type returned here? If lock fails that is problem it seems. I don't have any suggestions at the moment, but it does seem important.

Reply:
If we encounter a file where we're supposed to write we move it to a backup location. So the only failing I can see it doing is if things start going haywire with the OS' file system. I'm not sure if that's something we should encode into the return type, as everything starts breaking down at that point.